# exlude posix time_exploded and time_now from mac platform and replace waitable_event_posix with waitable_event_mac

--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -589,6 +589,12 @@ def write_gn_ninja(path, root_gen_dir, o
       'base/vlog.cc',
   ])
 
+  if is_posix and not is_mac:
+    static_libraries['base']['sources'].extend([
+        'base/time/time_exploded_posix.cc',
+        'base/time/time_now_posix.cc',
+    ])
+
   if is_posix:
     static_libraries['base']['sources'].extend([
         'base/base_paths_posix.cc',
@@ -613,7 +619,7 @@ def write_gn_ninja(path, root_gen_dir, o
         'base/synchronization/condition_variable_posix.cc',
         'base/synchronization/lock_impl_posix.cc',
         'base/synchronization/read_write_lock_posix.cc',
-        'base/synchronization/waitable_event_posix.cc',
+        'base/synchronization/waitable_event_mac.cc',
         'base/sys_info_posix.cc',
         'base/task_scheduler/task_tracker_posix.cc',
         'base/threading/platform_thread_internal_posix.cc',
@@ -621,8 +627,6 @@ def write_gn_ninja(path, root_gen_dir, o
         'base/threading/thread_local_storage_posix.cc',
         'base/threading/worker_pool_posix.cc',
         'base/time/time_conversion_posix.cc',
-        'base/time/time_exploded_posix.cc',
-        'base/time/time_now_posix.cc',
         'base/trace_event/heap_profiler_allocation_register_posix.cc',
     ])
     static_libraries['libevent'] = {
